plugins/chat/assets/javascripts/discourse/components/chat-thread.gjs

```diff
@@ -528,6 +544,108 @@ export default class ChatThread extends Component {
     return prev;
   }
 
+  get membership() {
+    return this.args.thread.currentUserMembership;
+  }
+
+  @action
+  createThreadMembership() {
+    if (this.membership || !this.currentUser.admin) {
+      return;
+    }
+
+    this.args.thread.currentUserMembership = UserChatThreadMembership.create({
+      notification_level: NotificationLevels.TRACKING,
+      last_read_message_id: this.args.thread.lastMessageId,
+      thread_title_prompt_seen: false,
+    });
+  }
+
+  @action
+  async updateThreadTitlePrompt() {
+    if (!this.membership) {
+      return;
+    }
+
+    try {
+      await this.chatApi.updateCurrentUserThreadTitlePrompt(
+        this.args.thread.channel.id,
+        this.args.thread.id,
+        { thread_title_prompt_seen: true }
+      );
+    } catch (e) {
+      // eslint-disable-next-line no-console
+      console.log("Couldn't save thread title prompt status", e);
+      this.membership.threadTitlePromptSeen = false;
```

Comment:
I would rather put this line in finally block

Reply:
I think we only want to populate this value if the request to chat api fails though

Reply:
ah ok ic :+1: